Automattic/woocommerce-payments
# refactor: separate BNPL methods from settings list

Fixes https://github.com/Automattic/woocommerce-payments/issues/8248

#### Changes proposed in this Pull Request









Separating BNPLs into their own section in the settings.
They're already separate in onboarding, so this should be the only area they need to be separated.

The component rendering the list of payment methods (`client/payment-methods/index.js`) has some specific logic to show a modal on deletion, show duplicate payment method information, and do some other business logic.

I wanted to ensure we could render the payment methods in two section using the same component, without duplicating the logic.

So I made the `client/payment-methods/index.js` component "reusable" by passing a list of payment methods to render. I moved it to `client/settings/payment-methods-list/index.js`. And I wrapped it by the newly created `BuyNowPayLaterSection` and `PaymentMethodsSection`.

There's some files moved around (mainly everything from `client/payment-methods` moved to `client/settings/payment-methods-list`). So that's why there's lots of changes.

After:
<img width="1083" alt="Screenshot 2024-08-15 at 1 23 16 PM" src="https://github.com/user-attachments/assets/d1dd42f2-bca2-413c-9f60-5d1dcdd73198">


#### Testing instructions





- Go to the settings page at https://wcpay.test/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments
- The BNPL methods should be rendered in a separate section
- Credit Card is still the first element on the list
- Un-checking a method shows the confirmation modal (on both BNPL and non-BNPL)
<img width="622" alt="Screenshot 2024-08-15 at 4 12 32 PM" src="https://github.com/user-attachments/assets/37937944-f5ee-4ca3-bb1e-b5a89a4abac3">


-------------------


**Post merge**
